Traducao managing pull requests (#54)
gatsbyjs/gatsby-pt-BR: docs/contributing/managing-pull-requests.md

---

> title: Gerencie Pull Requests

Reviewer: Gerencie -> Gerenciando
```suggestion
title: Gerenciando Pull Requests
```

Author: Eu acredito que os dois casos estão corretos, optei pelo primeiro pq não sou fã do gerúndio. De qualquer forma, aceito a sugestão.

---

Se você deseja ajudar a gerenciar PRs no repositório Gatsby no Github, este documento é para você. Abordaremos as convenções preferidas pela equipe, o que foi verificado em diversos tipos de pull requests, permissões, e orientações de como deixar um comentário. 

Temos mais de 2.000 colaboradores e muito do que torna o Gatsby excelente é contribuído por pessoas como você.

Será desnecessário dizer, temos muitos PRs e mergeamos mais de [100 contribuições](https://twitter.com/kylemathews/status/1111435640581689345) toda semana. Sim, _toda semana_.

Vamos conversar um pouco sobre como gerenciar pull requests no repositório Gatsby.

Para introduzir o que são Pull Requests e como abrir um, confira o documento de contribuição sobre [Como abrir um Pull Request](/contributing/how-to-open-a-pull-request).

## Verificando Pull Request

### Diretrizes Gerais

> Algumas coisas gerais para serem verificadas em um pull request:

Reviewer: Talvez fique melhor assim: 
```suggestion
Algumas coisas gerais para serem verificadas em um pull request são:
```

- Os links devem ser relativos, em vez de absolutos, ao vincular documentos (`/docs/some-reference/` ao invés de `https://www.gatsbyjs.org/docs/some-reference/`)
- A linguagem deve ser inclusiva e acessível
- Issues e RFCs (se tiver) devem estar vinculados ao endereço deste PR 

> > 💡 Ao olhar para um PR pela primeira vez, pode ajudar a ler sobre problemas vinculados ou [RFCs](/contributing/rfc-process/) (se houver) para entender o contexto sobre o que PR pretende adicionar ou corrigir.

Reviewer 2: ```suggestion
> 💡 Ao olhar para um PR pela primeira vez, pode ser de grande ajuda ler sobre problemas vinculados ou [RFCs](/contributing/rfc-process/) (se houver) para entender o contexto sobre o que o PR pretende adicionar ou corrigir.
```

Mesmo que não seja literal, a tradução fica melhor eu acho

### Tipos Específicos de Diretrizes

Cada tipo de PR também exige um conjunto diferente de verificações antes de serem mergeadas.

Vamos examiná-las abaixo.

#### Documentação

> Normalmente, procuramos o seguinte em [PRs que adicionam documentação](/contribuindo/docs-contribuicoes/): 

Reviewer: ```suggestion
Normalmente, procuramos o seguinte em [PRs que adicionam documentação](/contributing/docs-contributions/): 
```

Author: Fiquei na dúvida se deveria traduzir ou não as url's tbm

- Correção - se a documentação adicionada está tecnicamente correta

> - Estilo - se a linguagem escrita segue o nosso [guia de estilo](/contribuindo/guia-estilo-gatsby/)

Reviewer: ```suggestion
- Estilo - se a linguagem escrita segue o nosso [guia de estilo](/contributing/gatsby-style-guide/)
```

- Cabeçalhos – se os níveis de cabeçalho em um documento começam com h2 (`##` no Markdown) e crescem em ordem, estabelecendo uma hierarquia de conteúdo acessível

> - Tipo e Formato - se os documentos e os materiais de aprendizagem estão alinhados com nossas recomendações e [modelos de documentos](/contribuindo/modelos-documentos/) 

Reviewer: é importante mantermos os links não traduzidos pois eles fazem parte da documentação e são referenciados na hora quando são clicados.

```suggestion
- Tipo e Formato - se os documentos e os materiais de aprendizagem estão alinhados com nossas recomendações e [modelos de documentos](/contributing/docs-templates/) 
```

#### Código

> Para [PRs que adicionam código](/contribuindo/contribuicao-código/) (seja um recurso ou uma correção) procuramos o seguinte: 

Reviewer: Tente manter o link original traduzindo apenas a parte "clicável" do link 😉
```suggestion
Para [PRs que adicionam código](/contributing/code-contributions/) (seja um recurso ou uma correção) procuramos o seguinte: 
```

> Para [PRs que adicionam código](/contribuindo/contribuicao-código/) (seja um recurso ou uma correção) procuramos o seguinte: 

Reviewer: ```suggestion
Para [PRs que adicionam código](/contributing/code-contributions/) (seja um recurso ou uma correção) procuramos o seguinte: 
```

> - Correção — se o código faz o que pensamos que faz

Reviewer 2: ```suggestion
- Corretude — se o código faz o que pensamos que faz
```
Acho que essa é uma tradução melhor

> - Testes — ao corrigir um bug ou adicionar um novo recurso, pode ser muito valioso adicionar testes. Embora mergeamos alguns PRs pequenos sem teste, mais frequentemente do que não, é bom ter testes afirmando o comportamento. Pode ser uma combinação de testes de unitários para um pacote específico, testes de captura instantânea e testes de ponta a ponta. O objetivo aqui é garantir que algo que está sendo corrigido ou adicionado _permanece_ corrigigo ou funcione da maneira que é esperada. Bons testes garantem isso.

Reviewer 2: ```suggestion
- Testes — ao corrigir um bug ou adicionar um novo recurso, pode ser muito valioso adicionar testes. Embora mergeamos alguns PRs pequenos sem teste, mais frequentemente do que não, é bom ter testes afirmando o comportamento. Pode ser uma combinação de testes unitários para um pacote específico, testes de captura instantânea e testes de ponta a ponta. O objetivo aqui é garantir que algo que está sendo corrigido ou adicionado _permanece_ corrigido ou funcione da maneira que é esperada. Bons testes garantem isso.
```

" mais frequentemente do que não" tu acha que ficou estranho não?

> - Qualidade do código — concentre-se em alterações razoáveis que provavelmente melhorarão a manutenção, compreensão ou correção do código. Alterações estilísticas são tipicamente sugeridas por Prettier. Não escolha.

Reviewer: Acho que soa melhor se usarmos "pelo Prettier" em vez de "por Prettier"

```suggestion
- Qualidade do código — concentre-se em alterações razoáveis que provavelmente melhorarão a manutenção, compreensão ou correção do código. Alterações estilísticas são tipicamente sugeridas pelo Prettier. Não escolha.
```

- Documentação no README do pacote se você estiver adicionando algo

> #### Iniciantes ou Mostruários de Site

Reviewer: Esses _Starters_ são relativos a sites do Gatsby e não estão sendo traduzidos. Para palavras nesse sentido, dê uma olhada em nosso [glossário](https://docs.google.com/spreadsheets/d/1TLztI-jrPslCt0fgk4B9tGnhhS_7JQZJfMql1ePsIZo/edit#gid=0) 😉

```suggestion
#### Starters ou Mostruários de Site
```

> Para PRs que adicionam um site ou um iniciante de mostruário, devemos verificar:

Reviewer: ```suggestion
Para PRs que adicionam um site ou um starter ao mostruário, devemos verificar:
```

> - Verifique se o site ou o iniciador foi criado com Gatsby

Reviewer: ```suggestion
- Verifique se o site ou o starter foi criado com Gatsby
```

> - Links — verifique se os links estão funcionando e acessíveis

Reviewer 2: ```suggestion
- Links — verifique se os links estão acessíveis e funcionando
```
Talvez seja besteira minha, mas o que tu acha dessa sugestão?

Author: Acho ok

- Tags — verifique se as tags correspondem às tags existentes
- Status em destaque — novos sites não devem ser marcados em destaque. 

> Os sites em destaque são ocasionalmente atualizados por um mebro da equipe Gatsby.

Reviewer: ```suggestion
Os sites em destaque são ocasionalmente atualizados por um membro da equipe Gatsby.
```

#### Postagens no blog

Para PRs que adicionam postagens, devemos verificar: 

> - Correção — se a documentação adicionada está tecnicamente correta

Reviewer 2: ```suggestion
- Corretude — se a documentação adicionada está tecnicamente correta
```

> - Estilo — se a linguagem escrita segue o nosso [guia de estilo](/contribuindo/guia-estilo-gatsby/)

Reviewer: ```suggestion
- Estilo — se a linguagem escrita segue o nosso [guia de estilo](/contributing/gatsby-style-guide/)
```

> - Assunto — as postagens do blog não devem ser puramente promocionais com spam ou inapropriadas. Um autor deve verificar com um membro da equipe do Gatsby se sua postagem é apropriada para o blog antes de criar seu PR.

Reviewer 2: ```suggestion
- Assunto — as postagens do blog não devem ser puramente promocionais, com spam ou inapropriadas. Um autor deve verificar com um membro da equipe do Gatsby se sua postagem é apropriada para o blog antes de criar seu PR.
```

> - Sensibilidade ao tempo — as postagens do blog levam mais tempo do que os documentos, especialmente porque são enterradas após a publicação de mais postagens. Se algo é continuamente relevante e mais próximo de um tutorial genérico, provavelmente deve estar na seção [Guias de referência](/documentos/guias/) dos documentos.

Reviewer: ```suggestion
- Sensibilidade ao tempo — as postagens do blog levam mais tempo do que os documentos, especialmente porque são enterradas após a publicação de mais postagens. Se algo é continuamente relevante e mais próximo de um tutorial genérico, provavelmente deve estar na seção [Guias de referência](/docs/guides/) dos documentos.
```

## Verificações automáticas

> Nosso repositório no [GitHub](https://github.com/gatsbyjs/gatsby) possui várias verificações de IC automatizadas que são executadas automaticamente para todos os PRs. Isso inclui testes, linting e até visualizações para o [gatsbyjs.org](https://www.gatsbyjs.org).

Reviewer: Talvez seja interessante deixar como CI (Integração Contínua)
```suggestion
Nosso repositório no [GitHub](https://github.com/gatsbyjs/gatsby) possui várias verificações de _CI_ (Integração Contínua) automatizadas que são executadas automaticamente para todos os PRs. Isso inclui testes, _linting_ e até visualizações para o [gatsbyjs.org](https://www.gatsbyjs.org).
```

Author: Boa! Tinhado achado estranho tbm deixar só `IC`

> Nosso repositório no [GitHub](https://github.com/gatsbyjs/gatsby) possui várias verificações de _CI_ (Integração Contínua) automatizadas que são executadas automaticamente para todos os PRs. Isso inclui testes, _linting_ e até visualizações para o [gatsbyjs.org](https://www.gatsbyjs.org).

Reviewer 2: ```suggestion
Nosso repositório no [GitHub](https://github.com/gatsbyjs/gatsby) possui várias verificações de _CI_ (Integração Contínua) automatizadas que são executadas automaticamente para todos os PRs. Isso inclui testes, _linting_ e até pré-visualizações para o [gatsbyjs.org](https://www.gatsbyjs.org).
```

Queremos que todas essas verificações passem. Embora seja bom revisar um PR em andamento com algumas verificações com falha, um PR está pronto para ser enviado quando todos os testes forem aprovados.

> Vamos examinar alguns casos falhos comund e como corrigi-los:

Reviewer: ```suggestion
Vamos examinar alguns casos falhos comuns e como corrigi-los:
```

### Linting

> Conhecemos todo o código e documentação para obter consistência. Você pode achar que seu PR falhou na verificação do linting.

Reviewer: ```suggestion
Conhecemos todo o código e documentação para obter consistência. Você pode achar que seu PR falhou na verificação do _linting_.
```

> Conhecemos todo o código e documentação para obter consistência. Você pode achar que seu PR falhou na verificação do _linting_.

Reviewer 2: ```suggestion
Executamos o _lint_ em todo o código e documentação para obter consistência. Você pode acabar desconfiando que seu PR falhou na verificação do _linting_.
```
Acho que algo assim seja mais perto da tradução correta. O que tua acha?

> Se este for o seu PR e você tiver o código verificado em sua máquina, poderá executar:

Reviewer 2: ```suggestion
Caso o PR seja seu e você tiver o código em sua máquina, você pode executar:
```

```shell
npm run format
```

> Isso reformatará automaticamente suas alterações para atender aos requisitos do linting. Não se esqueça de fazer git commit e push com suas novas alterações.

Reviewer: ```suggestion
Isso reformatará automaticamente suas alterações para atender aos requisitos do _linting_. Não se esqueça de fazer git commit e push com suas novas alterações.
```

## Outras verificações

### Testando localmente

> Embora tenhamos muitos _muitos_ testes em nosso repositório (que são executados automaticamente em cada confirmação), pode haver momentos em que exista um caso específico (ou cinco) que não seja coberto por eles.

Reviewer: ```suggestion
Embora tenhamos muitos _muitos_ testes em nosso repositório (que são executados automaticamente em cada commit), pode haver momentos em que exista um caso específico (ou cinco) que não seja coberto por eles.
```

> Embora tenhamos muitos _muitos_ testes em nosso repositório (que são executados automaticamente em cada commit), pode haver momentos em que exista um caso específico (ou cinco) que não seja coberto por eles.

Reviewer 2: ```suggestion
Embora tenhamos muitos e _muitos_ testes em nosso repositório (que são executados automaticamente em cada commit), pode haver momentos em que exista um caso específico (ou cinco) que não seja coberto por eles.
```

Nessas situações, testar a alteração localmente pode ser muito valioso.

> > 💡 Caso seja a primeira vez que você faça isso, talvez seja necessário [configurar seu ambiente de desenvolvimento](/contribuir/configurar-seu-ambiente-de-desenvolvimento).

Reviewer: ```suggestion
> 💡 Caso seja a primeira vez que você faça isso, talvez seja necessário [configurar seu ambiente de desenvolvimento](/contributing/setting-up-your-local-dev-environment).
```

> Testar pacotes não publicados localmente pode ser complicado. Temos apenas a ferramenta para facilitar isso.

Reviewer 2: ```suggestion
Testar pacotes não publicados localmente pode ser complicado. Felizmente Temos a ferramenta certa para facilitar isso.
```

Acho que segue mais o sentido original

> Testar pacotes não publicados localmente pode ser complicado. Felizmente Temos a ferramenta certa para facilitar isso.

Reviewer 2: ```suggestion
Testar pacotes não publicados localmente pode ser complicado. Felizmente temos a ferramenta certa para facilitar isso.
```

Diga olá para o seu novo melhor amigo, `gatsby-dev-cli`.

#### gatsby-dev-cli

`gatsby-dev-cli` é uma ferramenta de linha de comando para o desenvolvimento local do Gatsby. Ao fazer alterações nos pacotes gatsby, isso ajuda a copiar as alterações nos pacotes para um site do Gatsby no qual você pode testar suas alterações.

Confira o arquivo [`gatsby-dev-cli` README](https://github.com/gatsbyjs/gatsby/tree/master/packages/gatsby-dev-cli) para saber mais.

### Commit e Título do PR

É bom ter mensagens de commit descritivas para que outras pessoas possam entender o que seu commit está fazendo. Gostamos de [commits convencionais](https://www.conventionalcommits.org/en/v1.0.0-beta.3).

> No entanto, os PRs são adicionados quando mergeados, de modo que as mensagens individuais de confirmação não são tão importantes quanto os títulos de PR.

Reviewer: ```suggestion
No entanto, os PRs são adicionados quando mergeados, de modo que as mensagens individuais de commit não são tão importantes quanto os títulos de PR.
```

Vejamos alguns exemplos de bons e ruins títulos de PR:

#### Bons títulos de PR ✅

> - tarefa(documentos): Corrija os links na página de contribuições

Reviewer 2: ```suggestion
- tarefa(documentos): Correção dos links na página de contribuições
```

> - feat(gatsby): Adicionar suporte para manifestos por página

Reviewer 2: ```suggestion
- feat(gatsby): adição do suporte para manifestos por página
```

> - correções(gatsby-plugin-sharp): Verifique se as imagens existem antes de tentar converter

Reviewer 2: ```suggestion
- correções(gatsby-plugin-sharp): Verificação das existência das imagens antes de tentar converter
```

Essas mudanças acima eu fiz porque como no original é um título de PR, faz sentido o título ter a correção que ele fez. A tua tradução tem um título que faz mais sentido para uma Issue

> Estes são bons títulos de PR porque são concisos, específicos e usam o formato [commit convencional](https://www.conventionalcommits.org/en/v1.0.0-beta.3).

Reviewer 2: ```suggestion
Estes são bons títulos de PR porque são concisos, específicos e usam o formato [commit convencional](https://www.conventionalcommits.org/pt-br/v1.0.0-beta.4/).
```

Tem a versão em pt do site :)

#### Títulos ruins de PR ❌

- novos testes
- adicionar suporte para meus novos cms
- correção de bug no gatsby

> Esses são títulos ruins de PR porque são genéricos, não comunicam a alteração corretamente e  não usam o formato de commit convencional.

Reviewer 2: ```suggestion
Esses são títulos ruins de PR porque são genéricos, não comunicam a alteração corretamente e não usam o formato de commit convencional.
```

> ## Dar Feedback

Reviewer 2: ```suggestion
## Fornecer Feedback
```

> - Seja _gentil_. Todos os dias somos mais fortes por causa da nossa comunidade, por isso a compaixão é importante. Queremos que todos os colaboradores se sintam bem-vindos. 

Reviewer 2: ```suggestion
- Seja _gentil_. Nos tornamos mais fortes a cada dia por causa da nossa comunidade, por isso a compaixão é importante. Queremos que todos os colaboradores se sintam bem-vindos. 
```
O que tua acha?

- Faça sugestões usando o [Recurso de sugestões do GitHub](https://help.github.com/en/articles/commenting-on-a-pull-request#adding-line-comments-to-a-pull-request), se possível. Isso facilita a aceitação de suas sugestões para o autor.
- Link para exemplos quando necessário.
- Tente não fazer [exageros](http://bikeshed.com/) demais.
- Observe quando uma sugestão é opcional (ao contrário de obrigatório)
- Seja objetivo e limite as remoções (alguns são bons se agregar valor ou melhorar a legibilidade do código)
- Não sugira nem espere mudanças fora do escopo que sejam melhor tratadas em um PR separado.

## Direitos e Permisões

### Quem pode revisar um PR?

Se você é um membro da organização [gatsbyjs](http://github.com/gatsbyjs) no GitHub, pode revisar um PR.

> > 💡 Ainda não é membro? Deseja [participar da contribuição](/contribuicao/como-contribuir/) para projetos de código aberto? Faça sua primeira contribuição e você será convidado automaticamentre!

Reviewer: ```suggestion
> 💡 Ainda não é membro? Deseja [participar da contribuição](/contributing/how-to-contribute/) para projetos open source? Faça sua primeira contribuição e você será convidado automaticamente!
```

### Quem pode aprovar um PR?

> Todo PR aberto no repositório precisa ser aprovado antes que possa  ser mergeado. Embora qualquer pessoa que seja membro da organização [gatsbyjs](http://github.com/gatsbyjs) possa aprovar um PR, para ser mergeado, ele precisa ser revisado por um membro da equipe Gatsby.

Reviewer 2: ```suggestion
Todo PR aberto no repositório precisa ser aprovado antes que possa ser mergeado. Embora qualquer pessoa que seja membro da organização [gatsbyjs](http://github.com/gatsbyjs) possa aprovar um PR, para ser mergeado, ele precisa ser revisado por um membro da equipe Gatsby.
```

Espaço desnecessário
